holocube/element/cube.py

```diff
@@ -0,0 +1,68 @@
+import iris
+import param
+import numpy as np
+
+from holoviews.core import Dimension, Element, HoloMap
+
+
+def coord_to_dimension(coord):
+    kwargs = {}
+    if coord.units.is_time_reference():
+        kwargs['value_format'] = coord.units.num2date
+    else:
+        kwargs['unit'] = str(coord.units)
+    return Dimension(coord.name(), **kwargs)
+
+
+class Cube(Element):
+    
+    def __init__(self, data, **params):
+        if isinstance(data, iris.cube.Cube):
+            if 'kdims' in params:
+                kdims = params['kdims']
+                if len(kdims) != len(data.dim_coords):
+                    raise ValueError('Supplied key dimensions must match Cube dim_coords.')
+                coords = []
+                for kd in kdims:
+                    coord = data.coords(kd.name if isinstance(kd, Dimension) else kd)
+                    if len(coord) == 0:
+                        raise ValueError('Key dimension %s not found in Iris cube.' % kd)
+                    coords.append(coord[0])
+            else:
+                coords = data.dim_coords
+            params['kdims'] = [coord_to_dimension(coord)
+                               for coord in coords]
+            if 'vdims' not in params:
+                params['vdims'] = [Dimension(data.name(), unit=str(data.units))]
+        else:
+            raise ValueError('Cube data must be of Iris Cube type.')
```

Comment:
I personally prefer to reverse the order of checks like this, checking datatype at the start and raising the error right away.  That way the rest of the function won't need indenting, which avoids requiring the user to put it all in a mental stack buffer.